### PR TITLE
refactor(main): move map_port to network

### PR DESCRIPTION
**Pull request by ReenigneArcher**
## Description

This PR moves the `map_port` function out of `main`, and into `network`.

Enabler for #1603 

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I want maintainers to keep my branch updated

### CODE REVIEW

File: src/network.cpp [hunk 1]
```diff
@@ -4,11 +4,37 @@
  */
 #include "network.h"
 #include "config.h"
+#include "logging.h"
 #include "utility.h"
 #include <algorithm>
```
Comment (ReenigneArcher) [inline src/network.cpp:9]: I'm not really sure why clang format wants `network.h` first, but whatever.

**Comment by ReenigneArcher** [comment]
@cgutman Done! Also found a `map_port` function in `upnp` so renamed that to avoid confusion (and a related function for consistency).

File: src/network.cpp (hunk 1 above)
Comment (cgutman) [reply src/network.cpp:9]: Maybe it's treated differently because it's the corresponding .h file for this .cpp file.

**Review by cgutman — APPROVED** [review]

File: src/network.cpp (hunk 1 above)
Comment (ReenigneArcher) [reply src/network.cpp:9]: I was thinking that too but didn't necessarily see the same behavior in other cpp files.

**Pull request merged**